### PR TITLE
Moodle Autoscale template 

### PR DESCRIPTION
**Pull request by CatalystAUInf**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Initial submission for pull request

### CODE REVIEW

**Comment by rgardler** [comment]
As the ReadMe says, this is an improvement on an existing Moodle template. Is there any reason why we should not merge this into the existing template rather than create a second one?

File: moodle-autoscale-caching/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+                "postgres",
+                "mysql"
+            ],
+            "metadata": {
+                "description": "Database type"
+            },
+            "type": "string"
+        },
+        "controllerVmSku": {
+            "defaultValue": "Standard_DS1",
```
[the remark this answers is not in the document]
Comment (CatalystAUInf) [reply moodle-autoscale-caching/azuredeploy.json]: This has been changed in commit 30890ec

File: moodle-autoscale-caching/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+            "blobStorageAccountName": "[tolower(concat('abs',variables('resourceprefix')))]",
+            "blobStorageAccountType": "[parameters('blobStorageAccountType')]",
+            "computeApi": "2016-04-30-preview",
+            "controllerVmSku": "[parameters('controllerVmSku')]",
+            "dbLogin": "[parameters('dbLogin')]",
+            "dbLoginPassword": "[concat(substring(uniqueString(resourceGroup().id, deployment().name), 2, 11), '*7', toUpper('pfiwb'))]",
+            "dbServerType": "[parameters('dbServerType')]",
+            "dbUsername": "[concat(parameters('dbLogin'), '@', parameters('dbServerType'), '-', variables('resourceprefix'))]",
+            "elasticAvailabilitySetName": "[concat('elastic-avset-',variables('resourceprefix'))]",
+            "elasticClusterName": "[concat('es-cluster-',variables('resourceprefix'))]",
```
[the remark this answers is not in the document]
Comment (CatalystAUInf) [reply moodle-autoscale-caching/azuredeploy.json]: This has been added in commit 30890ec

**Comment by rgardler** [comment]
When deployment is completed the LB DNS is set to "lb-UNIQUESTRING.LOCATION.cloudapp.azure.com" but Moodle is configured with the name provided in the parameter `siteURL` in azuredeploy.parameters. Moodle redirects to this URL when using the LB DNS name. 

Since the customer does not know what (to be generated) "UNIQUESTRING" will be the LB DNS how are they supposed to configure siteURL in a test scenario? Is your expectation that users will already have a DNS registration that they can direct to the LB?

**Comment by CatalystAUInf** [comment]
> Since the customer does not know what (to be generated) "UNIQUESTRING" will be the LB DNS how are they supposed to configure siteURL in a test scenario? Is your expectation that users will already have a DNS registration that they can direct to the LB?

This is correct. Moodle requires a specific hostname to function. If you put the LB dns address as the site url, then pointing an actual url (www.example.org) to that will either redirect to https://<load balancer name> or display a broken Moodle. It is assumed that the siteurl will be configured either as part of a deploy with parameters or by entering in the desired site name when launching the stack.

The load balancer DNS name is presented as an output so that users can CNAME their desired url to the load balancer. All the outputs have been listed in the README.md file.

**Comment by rgardler** [comment]
OK, that's not really an acceptable behavior. The goal of the Quickstarts is for them to be, well Quick. That said, for real use it is important to be able to configure the siteURL as currently implemented. In my [fork of your repo](https://github.com/rgardler/azure-quickstart-templates/tree/catalyst-autoscale-addition) I've added a feature such that if `siteURL` is omitted or it is left as the default ("www.example.org") it will be overwritten with the generated DNS. This allows both "quickstart" and the "production" behavior.

I also think this work should replace the existing Moodle template.

For this reason I will not be merging this, but will issue a separate PR from my fork of your work. So while I am going to close this PR I will be merging your work soon.

**Comment by rgardler** [comment]
Closing as this patch is included in #4173 and will be merged as part of that work.

**Pull request closed without merging**